File: Delphi/Modules/20.Reports/SharedData/DemoData.pas
unit DemoData;

interface

uses
  SysUtils, Classes, ADODB, DB;

type
  TDemoTables = class(TDataModule)
    ADOConnection: TADOConnection;
    Categories: TADODataSet;
    Products: TADODataSet;
    DsCategories: TDataSource;
    OrderDetails: TADODataSet;
    DsProducts: TDataSource;
    Employees: TADODataSet;
    Orders: TADODataSet;
    DsEmployees: TDataSource;
    Suppliers: TADODataSet;
    Shippers: TADODataSet;
    procedure DataModuleCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  DemoTables: TDemoTables;

implementation
uses IOUtils;

{$R *.dfm}

function DBFile: string;
begin
  Result := TPath.Combine(TPath.GetDirectoryName(ParamStr(0)), '..\..\..\SharedData\Northwind.mdb');
end;

procedure TDemoTables.DataModuleCreate(Sender: TObject);
begin
  ADOConnection.ConnectionString := StringReplace(ADOConnection.ConnectionString, 'Northwind.mdb', DbFile, []);
end;

end.

File: Cpp/Modules/20.Reports/SharedData/DemoData.cpp
//---------------------------------------------------------------------------


#pragma hdrstop

#include "DemoData.h"
#include <IOUtils.hpp>
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma classgroup "Vcl.Controls.TControl"
#pragma resource "*.dfm"
TDemoTables *DemoTables;
//---------------------------------------------------------------------------
__fastcall TDemoTables::TDemoTables(TComponent* Owner)
	: TDataModule(Owner)
{
}
//---------------------------------------------------------------------------

String DBFile()
{
  return TPath::Combine(TPath::GetDirectoryName(ParamStr(0)), "..\\..\\..\\SharedData\\Northwind.mdb");
};


void __fastcall TDemoTables::DataModuleCreate(TObject *Sender)
{
  ADOConnection->ConnectionString = StringReplace(ADOConnection->ConnectionString, "Northwind.mdb", DBFile(), TReplaceFlags());
}
//---------------------------------------------------------------------------

